_posts/2025-09-30-introduction-to-ai-agents-in-opensearch.md

```diff
@@ -0,0 +1,97 @@
+
+In the 2.12 release, OpenSearch introduced [Memory APIs](https://docs.opensearch.org/latest/ml-commons-plugin/api/memory-apis/index/), which store raw conversation messages and records of tools used in OpenSearch indexes to support conversational history retrieval and tracing. This Memory API preserves the messages and tool usage in their raw form, serving as a type of short-term memory.
+
+Recently, OpenSearch expanded to also support long-term memory, with the addition of _agentic memory_. Introduced in [OpenSearch 3.2](https://opensearch.org/blog/introducing-opensearch-3-2-next-generation-search-and-anayltics-with-enchanced-ai-capabilities/) and generally available in OpenSearch 3.3, agentic memory is a persistent memory system that enables AI agents to learn, remember, and reason across conversations and interactions. This feature provides comprehensive memory management using multiple strategies, including semantic fact extraction, user preference learning, and conversation summarization. It allows agents to maintain context and build knowledge over time and enables sophisticated memory operations, such as consolidation, retrieval, and history tracking. 
+
+By providing agents with persistent, searchable memory, this feature transforms static AI interactions into dynamic, context-aware experiences that improve over time, enabling more personalized and intelligent responses while maintaining control over memory organization and retention policies. For more information, see [Agentic memory](https://docs.opensearch.org/latest/ml-commons-plugin/agentic-memory/).
+
+## An agentic experience with an MCP server
+
+External agents can also benefit from accessing OpenSearch capabilities exposed as tools over standard MCP. OpenSearch offers two comprehensive [MCP server solutions](https://opensearch.org/blog/introducing-mcp-in-opensearch/) to meet diverse integration needs:
+ 
+1. **Built-in MCP Server**: The [built-in MCP server](https://docs.opensearch.org/latest/ml-commons-plugin/api/mcp-server-apis/index/) provides native integration within OpenSearch, exposing a streamable HTTP MCP API that enables dynamic tool registration and removal. This solution offers seamless access to core OpenSearch capabilities, allowing agents to invoke powerful [tools](https://docs.opensearch.org/latest/ml-commons-plugin/agents-tools/tools/index/) for real-time data queries and analysis. Since it makes direct API calls to OpenSearch through `/_plugins/_ml/mcp`, it natively supports Role-Based Access Control (RBAC), ensuring secure and granular access management.
```

Comment:
```suggestion
1. **Built-in MCP server**: The [built-in MCP server](https://docs.opensearch.org/latest/ml-commons-plugin/api/mcp-server-apis/index/) provides native integration within OpenSearch, exposing a Streamable HTTP MCP API that enables dynamic tool registration and removal. This solution offers seamless access to core OpenSearch capabilities, allowing agents to invoke powerful [tools](https://docs.opensearch.org/latest/ml-commons-plugin/agents-tools/tools/index/) for real-time data queries and analysis. Since it makes direct API calls to OpenSearch through `/_plugins/_ml/mcp`, it natively supports Role-Based Access Control (RBAC), ensuring secure and granular access management.
```